tests/scalers/splunk_observability_test/splunk_observability_test.go

```diff
@@ -0,0 +1,185 @@
+//go:build e2e
+// +build e2e
+
+package splunk_observability_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+const (
+	testName = "splunk-observability-test"
+)
+
+var (
+	testNamespace          = fmt.Sprintf("%s-ns", testName)
+	deploymentName         = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName       = fmt.Sprintf("%s-so", testName)
+	authName               = fmt.Sprintf("%s-auth", testName)
+	duration               = "10"
+	maxReplicaCount        = 10
+	minReplicaCount        = 1
+	scaleInTargetValue     = "400"
+	scaleInActivationValue = "1.1"
+)
+
+type templateData struct {
+	TestNamespace         string
+	DeploymentName        string
+	ScaledObjectName      string
+	AuthName              string
+	Duration              string
+	MinReplicaCount       string
+	MaxReplicaCount       string
+	TargetValue           string
+	ActivationTargetValue string
+}
+
+const (
+	authTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: splunk-secrets
+  namespace: {{.TestNamespace}}
+data:
+  accessToken: YW1JeUpqVHRJd185cDhOWG01X21KQQ==  # one time through-away access token used just for testing
```

Comment:
It seems this test relies on an actual upstream being available to communicate with. Would it be possible to create a pod here that simply mocks responses? We could override the endpoint in the scaler config and point it to our mocked API. This would ensure the tests could still run in a more closed loop fashion without any upstream dependencies. Thoughts?

Reply:
agree

Reply:
I just oriented myself on the other scaler e2e tests who are loading the credentials from an .env file, and I did the same.